rcl/src/rcl/init.c

```diff
@@ -144,6 +153,72 @@ rcl_init(
   rcutils_atomic_store((atomic_uint_least64_t *)(&context->instance_id_storage), next_instance_id);
   context->impl->init_options.impl->rmw_init_options.instance_id = next_instance_id;
 
+  size_t * domain_id = &context->impl->init_options.impl->rmw_init_options.domain_id;
+  if (RCL_DEFAULT_DOMAIN_ID == *domain_id) {
+    // Get actual domain id based on environment variable.
+    ret = rcl_get_default_domain_id(domain_id);
+    if (RCL_RET_OK != ret) {
+      fail_ret = ret;
+      goto fail;
+    }
+  }
+
+  rmw_localhost_only_t * localhost_only =
+    &context->impl->init_options.impl->rmw_init_options.localhost_only;
+  if (RMW_LOCALHOST_ONLY_DEFAULT == *localhost_only) {
+    // Get actual localhost_only value based on environment variable, if needed.
+    ret = rcl_get_localhost_only(localhost_only);
+    if (RCL_RET_OK != ret) {
+      fail_ret = ret;
+      goto fail;
+    }
+  }
+
+  if (context->global_arguments.impl->security_context) {
+    context->impl->init_options.impl->rmw_init_options.security_context = rcutils_strdup(
+      context->global_arguments.impl->security_context,
+      context->impl->allocator);
+  } else {
+    context->impl->init_options.impl->rmw_init_options.security_context = rcutils_strdup(
+      "/", context->impl->allocator);
+  }
+  int validation_result;
+  size_t invalid_index;
+  ret = rcl_validate_security_context_name(
+    context->impl->init_options.impl->rmw_init_options.security_context,
+    &validation_result,
+    &invalid_index);
+  if (RCL_RET_OK != ret) {
+    RCL_SET_ERROR_MSG("rcl_validate_security_context_name() failed");
+    fail_ret = ret;
+    goto fail;
+  }
+  if (RCL_SECURITY_CONTEXT_NAME_VALID != validation_result) {
+    RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
+      "security context name is not valid: '%s'. Invalid index: %zu",
```

Comment:
@ivanpauno nit^N: capitalize, as in 
```suggestion
      "Security context name is not valid: '%s'. Invalid index: %zu",
```

Reply:
See https://github.com/ros2/rcl/pull/515/commits/44a15be05484088c78a4e7cde08a021356757160.